packages/app/src/cli/services/scaffold/extension.ts

```diff
@@ -109,38 +105,29 @@ async function uiExtensionInit({
         title: `Scaffold ${getExtensionOutputConfig(extensionType).humanKey} extension`,
         task: async (_, task) => {
           task.title = `Scaffolding ${getExtensionOutputConfig(extensionType).humanKey} extension...`
-          const input = yaml.encode({
-            extensions: [
-              {
-                title: name,
-                // Use the new templates
-                external_type: mapExtensionTypeToExternalExtensionType(extensionType),
-                type: extensionType,
-                metafields: [],
-                development: {
-                  root_dir: '.',
-                  template: extensionFlavor,
-                  install_dependencies: false,
-                },
-              },
-            ],
-          })
-          await runGoExtensionsCLI(['create', '-'], {
-            cwd: extensionDirectory,
-            stderr: new stream.Writable({
-              write(chunk, encoding, next) {
-                task.output = chunk.toString()
-                next()
-              },
-            }),
-            stdout: new stream.Writable({
-              write(chunk, encoding, next) {
-                task.output = chunk.toString()
-                next()
-              },
-            }),
-            input,
+
+          const templateDirectory = await path.findUp(
+            `templates/ui-extensions/projects/${mapExtensionTypeToExternalExtensionType(extensionType)}`,
+            {
+              type: 'directory',
+              cwd: path.moduleDirectory(import.meta.url),
+            },
+          )
+
+          if (!templateDirectory) {
+            throw new error.Bug(`Couldn't find the template for ${extensionType}`)
+          }
+
+          await template.recursiveDirectoryCopy(templateDirectory, extensionDirectory, {
```

Comment:
It absolutely thrills me to see `recursiveDirectoryCopy` get used as a pattern again and again and again. Such a useful convention!